prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -87,36 +91,44 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
             // run our call to the endpoint in a blocking fashion
```

Comment:
Love using the httpClient as an argument passing to Class, this introduces IoC and Separates of Concern.  Therefore, making the code easy to test.